pyaedt/modeler/stackup_3d.py

```diff
@@ -0,0 +1,2284 @@
+        material : str, optional
+            Padstack Material
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.stackup_3d.Padstack`
+        """
+        p = Padstack(self._app, self, name, material)
+        self._padstacks.append(p)
+        return p
+
+    @pyaedt_function_handler()
+    def add_layer(self, name, layer_type="S", material="copper", thickness=0.035, fill_material="FR4_epoxy"):
+        """Add a new layer to the stackup. Can be Signal (S), ground (G) or Dielectric (D).
+        The layer is entirely filled with  fill_material. Anything will be drawn wmaterial.
+
+        Parameters
+        ----------
+        name : str
+            Layer name.
+        layer_type : str, optional
+            Layer type.Can be `"S"`, `"D"` or `"G"`.
+        material : str
+            Material name. Material will be parametrized.
+        thickness : float
+            Thickness value. Thickness will be parametrized.
+        fill_material : str
+            Fill Material name. Material will be parametrized. Not valid for Dielectrics.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.stackup_3d.Layer3D`
+            Layer Object.
+        """
+        self._shifted_index += 1
+        if not layer_type:
+            raise ValueError("Layer Type has to be one of the S, D, G strins.")
+        self._layer_name.append(name)
+
+        lay = Layer3D(
+            stackup=self,
+            app=self._app,
+            name=name,
+            layer_type=layer_type,
+            material=material,
+            thickness=thickness,
+            fill_material=fill_material,
+            index=self._shifted_index,
+        )
+        self._layer_position_manager(lay)
+        if layer_type == "D":
+            self._dielectric_list.extend(lay._obj_3d)
+            self._dielectric_name_list.append(lay._name)
+            lay._obj_3d[-1].transparency = "0.8"
+        elif layer_type == "G":
+            self._ground_list.extend(lay._obj_3d)
+            self._ground_name_list.append(lay._name)
+            self._ground_fill_material.append(lay._fill_material)
+            lay._obj_3d[-1].transparency = "0.6"
+            lay._obj_3d[-1].color = (255, 0, 0)
+
+        elif layer_type == "S":
+            self._signal_list.extend(lay._obj_3d)
+            self._signal_name_list.append(lay._name)
+            self._signal_material.append(lay._material_name)
+            # With the function _layer_position_manager i think this part is not needed anymore or has to be reworked
+            lay._obj_3d[-1].transparency = "0.8"
+        self._stackup[lay._name] = lay
+        return lay
+
+    @pyaedt_function_handler()
+    def add_signal_layer(self, name, material="copper", thickness=0.035, fill_material="FR4_epoxy"):
+        """Add a new ground layer to the stackup. A signal layer is positive.
+        The layer is entirely filled with  fill_material. Anything will be drawn wmaterial.
+
+        Parameters
```

Comment:
I don't see the function signature here--so I can't tell what parameters are optional and what the defaults for these are.